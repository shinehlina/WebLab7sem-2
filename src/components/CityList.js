import React from "react";
import City from "./City";
import { Preloader } from "react-materialize";
import Row from "react-materialize/lib/Row";
import Col from "react-materialize/lib/Col";

class CityList extends React.Component {
  onBtnClick = e => {
    if (e.keyCode === 13) {
      const city = e.target.value;
      if (
        !this.props.favoriteCities
          .map(c => c.cityName.toUpperCase())
          .includes(city.toUpperCase())
      ) {
        this.props.addCity(city);
      }
      e.target.value = "";
      e.preventDefault();
    }
  };

  createList(props) {
    if (props.favoriteCities.length === 0) {
      return <div>No favorite cities</div>;
    }
    let cityNameList = [];
    for (let i = 0; i < props.favoriteCities.length; i++) {
      let cityInfo = props.favoriteCities[i];
      if (cityInfo !== null) {
        cityNameList.push(
          <li key={cityInfo.cityName}>
            <City
              index={i}
              cityName={cityInfo.cityName}
              longtitude={cityInfo.data.coord.lon}
              latitude={cityInfo.data.coord.lat}
              description={cityInfo.data.weather[0].description}
              windSpeed={cityInfo.data.wind.speed}
              temperature={cityInfo.data.main.temp}
              humidity={cityInfo.data.main.humidity}
              pressure={cityInfo.data.main.pressure}
              deleteCity={props.deleteCity}
            />
          </li>
        );
      }
    }
    return cityNameList;
  }

  render() {
    return (
      <div>
        <Row>
          <Col className="valign-wrapper" s={6}>
            <h5 className="">Favorites</h5>
          </Col>
          <Col className="valign-wrapper" s={6}>
            <form>
              <input
                id="city"
                type="text"
                size="40"
                placeholder="Enter your city"
                onKeyDown={this.onBtnClick}
              />
            </form>
          </Col>
        </Row>
        {this.props.isFetching ? (
          <Preloader size="big" />
        ) : (
          <ul>{this.createList(this.props)}</ul>
        )}
      </div>
    );
  }
}

export default CityList;
