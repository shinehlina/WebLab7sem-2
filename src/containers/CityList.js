import React from "react";

class CityList extends React.Component {
  //   rendenCity(name, temperature) {
  //     return <City cityName={name} temperature={temperature} />;
  //   }

  onBtnClick = e => {
    if (e.keyCode === 13) {
      const city = e.target.value;
      this.props.addCity(city);
      e.target.value = "";
      e.preventDefault();
    }
  };

  createList(props) {
    let cityNameList = [];
    for (let i = 0; i < props.favoriteCities.length; i++) {
      let cityName = props.favoriteCities[i];
      if (cityName !== null) {
        // list.push(this.rendenCity(city.name, city.temperature));
        cityNameList.push(<div key={i}>{cityName}</div>);
      }
    }
    if (cityNameList.length === 0) {
      cityNameList.push(<div>No favorite cities</div>);
    }
    return cityNameList;
  }

  render() {
    return (
      <div>
        <h1> Города </h1>
        <form>
          <input
            id="city"
            type="text"
            size="40"
            placeholder="Enter your city"
            onKeyDown={this.onBtnClick}
          />
        </form>
        {this.createList(this.props)}
      </div>
    );
  }
}

export default CityList;
