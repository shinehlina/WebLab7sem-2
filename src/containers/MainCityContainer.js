import React from "react";
import MainCity from "../components/MainCity";
import { connect } from "react-redux";
import { getMainCity } from "../actions/mainLocationCity";

class MainCityContainer extends React.Component {
  render() {
    return (
      <MainCity
        getMainCity={this.props.getMainCityAction}
        data={this.props.cityData}
      />
    );
  }
}
const mapStateToProps = store => {
  return {
    cityData: store.mainCity
  };
};

const mapDispatchToProps = dispatch => {
  return {
    getMainCityAction: () => dispatch(getMainCity())
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MainCityContainer);
