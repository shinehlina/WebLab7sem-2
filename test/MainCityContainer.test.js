import React from "react";
import MainCityContainer from "../src/containers/MainCityContainer";
import { shallow } from "./enzyme";
import { shallowToJson } from "enzyme-to-json";
import configureStore from "redux-mock-store";
import { favoriteCities } from "./CityList.test";

const initialState = {};
const mockStore = configureStore();

describe("MainCityContainer didn't change ", function() {
  it("Empty storage", () => {
    const storage = [];
    const store = mockStore(initialState);
    const tree = shallow(<MainCityContainer value={storage} store={store} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it("Storage has elements", () => {
    const storage = favoriteCities;
    const store = mockStore(initialState);
    const tree = shallow(<MainCityContainer value={storage} store={store} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
