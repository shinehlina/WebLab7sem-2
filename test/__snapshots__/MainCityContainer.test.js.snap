// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityList error has cities didn't changed 1`] = `
<div>
  <Modal
    actions={
      Array [
        <Button
          flat={true}
          modal="close"
          node="button"
          waves="green"
        >
          Close
        </Button>,
      ]
    }
    bottomSheet={false}
    fixedFooter={false}
    header="Error"
    id="modal-0"
    open={true}
    options={
      Object {
        "dismissible": true,
        "endingTop": "10%",
        "inDuration": 250,
        "onCloseEnd": null,
        "onCloseStart": null,
        "onOpenEnd": null,
        "onOpenStart": null,
        "opacity": 0.5,
        "outDuration": 250,
        "preventScrolling": true,
        "startingTop": "4%",
      }
    }
  >
    Error
     
  </Modal>
</div>
`;

exports[`CityList error no cities didn't changed 1`] = `
<div>
  <Modal
    actions={
      Array [
        <Button
          flat={true}
          modal="close"
          node="button"
          waves="green"
        >
          Close
        </Button>,
      ]
    }
    bottomSheet={false}
    fixedFooter={false}
    header="Error"
    id="modal-0"
    open={true}
    options={
      Object {
        "dismissible": true,
        "endingTop": "10%",
        "inDuration": 250,
        "onCloseEnd": null,
        "onCloseStart": null,
        "onOpenEnd": null,
        "onOpenStart": null,
        "opacity": 0.5,
        "outDuration": 250,
        "preventScrolling": true,
        "startingTop": "4%",
      }
    }
  >
    Error
     
  </Modal>
</div>
`;

exports[`CityList loading didn't changed 1`] = `
<div>
  <Row>
    <Col
      className="valign-wrapper"
      s={6}
    >
      <h5
        className=""
      >
        Favorites
      </h5>
    </Col>
    <Col
      className="valign-wrapper"
      s={6}
    >
      <form>
        <input
          id="city"
          onKeyDown={[Function]}
          placeholder="Enter your city"
          size="40"
          type="text"
        />
      </form>
    </Col>
  </Row>
  <div
    className="center-align"
  >
    <Preloader
      active={true}
      color="blue"
      flashing={false}
      size="big"
    />
  </div>
</div>
`;

exports[`CityList success didn't changed 1`] = `
<div>
  <Row>
    <Col
      className="valign-wrapper"
      s={6}
    >
      <h5
        className=""
      >
        Favorites
      </h5>
    </Col>
    <Col
      className="valign-wrapper"
      s={6}
    >
      <form>
        <input
          id="city"
          onKeyDown={[Function]}
          placeholder="Enter your city"
          size="40"
          type="text"
        />
      </form>
    </Col>
  </Row>
  <Row>
    <Col
      key="Kiev"
      m={6}
      s={12}
    >
      <City
        cityName="Kiev"
        deleteCity={[Function]}
        index={0}
      />
    </Col>
  </Row>
</div>
`;

exports[`MainCityContainer didn't change  Empty storage 1`] = `
<ContextProvider
  value={null}
>
  <MainCityContainer
    getMainCityAction={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    value={Array []}
  />
</ContextProvider>
`;

exports[`MainCityContainer didn't change  Storage has elements 1`] = `
<ContextProvider
  value={null}
>
  <MainCityContainer
    getMainCityAction={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    value={
      Array [
        Object {
          "cityName": "Kiev",
          "data": Object {
            "coord": Object {
              "lat": 60.03,
              "lon": 30.33,
            },
            "main": Object {
              "humidity": 93,
              "pressure": 1028,
              "temp": 278.02,
            },
            "weather": Array [
              Object {
                "description": "light rain",
                "icon": "1",
                "main": "Rain",
              },
            ],
            "wind": Object {
              "speed": 3,
            },
          },
        },
      ]
    }
  />
</ContextProvider>
`;
